Source/Core/Core/AchievementManager.cpp

```diff
@@ -582,7 +581,8 @@ void AchievementManager::LeaderboardEntriesCallback(int result, const char* erro
     memcpy(map_entry.score.data(), response_entry.display, FORMAT_SIZE);
     map_entry.rank = response_entry.rank;
   }
-  AchievementManager::GetInstance().m_update_callback({.leaderboards = {leaderboard_id}});
+  AchievementManager::GetInstance().m_update_callback({.leaderboards = {*leaderboard_id}});
+  delete leaderboard_id;
```

Comment:
This will leak if the result is not `RC_OK`.

At the top of the function (before the if) you can do this:

```
Common::ScopeGuard on_end_scope([&]()
{
  delete userdata;
});
```